### PR TITLE
Show an error for webext -> legacy downgrades.

### LINKED ISSUE #3061: Don't allow developers to add a XUL add-on as an update to a webextension
In https://bugzilla.mozilla.org/show_bug.cgi?id=1283443, there was a discussion about how versioncheck doesn't work when a webextension is updated with a XUL add-on. Since this is something we don't plan to support, we can warn the developer from doing it during the submission process.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #3061

This also improves tests and fixes validation modification for the
previous commit related to upgrades.

screenshot from 2016-09-16 10-45-32

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
cc @jvillalobos or @atsay to review if the wording is fine.

**Comment by atsay** [comment]
“You cannot update a WebExtensions add-on with a legacy add-on. Your users would not be able to use your new version because Firefox does not support this type of update."

**Comment by jvillalobos** [comment]
Amy and I agreed on the text in the last comment, btw.

**Comment by EnTeQuAk** [comment]
Thanks for the input! Updated the text but I'm too lazy to update the screenshot :)

File: src/olympia/devhub/tasks.py [hunk 1]
```diff
@@ -281,6 +292,21 @@ def annotate_upgrade_warning(results, file_, addon, version_string):
             'message': msg,
             'description': [],
             'compatibility_type': None})
+        results['warnings'] += 1
```
Comment (eviljeff) [inline src/olympia/devhub/tasks.py]: is `results['warnings']` guaranteed to be defined, even when it's `0`?  (`+=` needs the value to be already defined, right?)

File: src/olympia/devhub/tasks.py [hunk 2]
```diff
@@ -281,6 +292,21 @@ def annotate_upgrade_warning(results, file_, addon, version_string):
...
+
+        messages = results['messages']
+        messages.insert(0, {
+            'tier': 1,
+            'type': 'error',
+            'id': ['validation', 'messages', 'webext_downgrade'],
+            'message': msg,
+            'description': [],
+            'compatibility_type': None})
+        results['errors'] += 1
```
Comment (eviljeff) [inline src/olympia/devhub/tasks.py]: same here

File: src/olympia/devhub/tests/test_tasks.py [hunk 3]
```diff
@@ -656,6 +658,12 @@ def test_webextension_webext_to_webext(self):
         upload.refresh_from_db()
 
         assert 'is_upgrade_to_webextension' not in upload.processed_validation
+        expected = ['validation', 'messages', 'webext_upgrade']
+        assert not any(
+            msg['id'] == expected
+            for msg in upload.processed_validation['messages'])
```
Comment (eviljeff) [inline src/olympia/devhub/tests/test_tasks.py]: Save upload.processed_validation in a variable so it's not being calculated numerous times?  (the property isn't cached afaik)
Same other in other places.

File: src/olympia/devhub/tasks.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/devhub/tasks.py]: Yes, it's guaranteed that it's defined. `devhub.utils:fix_addons_linter_output` set's it explicitly and relies on it too

File: src/olympia/devhub/tests/test_tasks.py (hunk 3 above)
Comment (EnTeQuAk) [inline src/olympia/devhub/tests/test_tasks.py]: Great idea, fixed in 2af8eb0

**Pull request merged**